.dotnet/src/Custom/Chat/ChatCompletionOptions.cs

```diff
@@ -1,113 +1,103 @@
+    /// </summary>
+    [CodeGenMember("Model")]
+    internal InternalCreateChatCompletionRequestModel Model { get; set; }
 
-    internal IDictionary<string, int> GetInternalLogitBias()
-    {
-        ChangeTrackingDictionary<string, int> packedLogitBias = [];
-        foreach (KeyValuePair<int, int> pair in TokenSelectionBiases)
-        {
-            packedLogitBias[$"{pair.Key}"] = pair.Value;
-        }
-        return packedLogitBias;
-    }
+    // CUSTOM: Made internal. We only ever request a single choice.
+    /// <summary> How many chat completion choices to generate for each input message. Note that you will be charged based on the number of generated tokens across all of the choices. Keep `n` as `1` to minimize costs. </summary>
+    [CodeGenMember("N")]
+    internal int? N { get; set; }
 
-    internal IList<Internal.Models.ChatCompletionTool> GetInternalTools()
-    {
-        ChangeTrackingList<Internal.Models.ChatCompletionTool> internalTools = [];
-        foreach (ChatToolDefinition tool in Tools)
-        {
-            if (tool is ChatFunctionToolDefinition functionTool)
-            {
-                FunctionDefinition functionObject = new(
-                    functionTool.Description,
-                    functionTool.FunctionName,
-                    functionTool.Parameters,
-                    serializedAdditionalRawData: null);
-                internalTools.Add(new(functionObject));
-            }
-        }
-        return internalTools;
-    }
+    // CUSTOM: Made internal. We set this manually based on the client method that is called.
+    /// <summary> If set, partial message deltas will be sent, like in ChatGPT. Tokens will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#Event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message. [Example Python code](https://cookbook.openai.com/examples/how_to_stream_completions). </summary>
+    [CodeGenMember("Stream")]
+    internal bool? Stream { get; set; }
 
-    internal IList<Internal.Models.ChatCompletionFunctions> GetInternalFunctions()
-    {
-        ChangeTrackingList<Internal.Models.ChatCompletionFunctions> internalFunctions = new();
-        foreach (ChatFunctionDefinition function in Functions)
-        {
-            Internal.Models.ChatCompletionFunctions internalFunction = new(
-                function.Description,
-                function.FunctionName,
-                CreateInternalFunctionParameters(function.Parameters),
-                serializedAdditionalRawData: null);
-            internalFunctions.Add(internalFunction);
-        }
-        return internalFunctions;
-    }
+    // TODO: What do we do about this?
```

Comment:
IMO we should initialize it and always set the "stream usage" to true. It feels exactly like the embeddings encoding format thing -- they couldn't make it the default out of an abundance of caution for breaking changes, but for libraries there's no good reason not to opt in.

Reply:
I added this!